File: sdk/kotlin/src/main/kotlin/com/litebase/client/Litebase.kt
package com.litebase.client

/**
 * Litebase — Kotlin client for Litebase Realtime.
 *
 * Mirrors the TypeScript `@litebase/client` and Swift `Litebase` SDKs:
 *
 * ```kotlin
 * import com.litebase.client.createClient
 *
 * val client = createClient(url = "http://127.0.0.1:3000")
 * val tokens = client.auth.signUp(email = "a@b.com", password = "secret12")
 * val list = client.from("todos").select(
 *     where = mapOf("completed" to WhereValue.Bool(false)),
 *     limit = 20,
 * )
 * val unsub = client.from("todos").subscribeReady { event ->
 *     println("${event.op} ${event.data}")
 * }
 * unsub()
 * client.closeRealtime()
 * ```
 */

data class LitebaseClientOptions(
    val url: String,
    val token: String? = null,
    val refreshToken: String? = null,
    val http: HttpTransport = OkHttpHttpTransport(),
    val webSocketFactory: WebSocketFactory? = null,
)

/** Create a Litebase client (TypeScript `createClient` equivalent). */
fun createClient(
    url: String,
    token: String? = null,
    refreshToken: String? = null,
    http: HttpTransport = OkHttpHttpTransport(),
    webSocketFactory: WebSocketFactory? = null,
): LitebaseClient {
    return LitebaseClient(
        LitebaseClientOptions(
            url = url,
            token = token,
            refreshToken = refreshToken,
            http = http,
            webSocketFactory = webSocketFactory,
        ),
    )
}

File: sdk/swift/Sources/Litebase/Litebase.swift
/**
 Litebase — Swift client for Litebase Realtime.

 Mirrors the TypeScript `@litebase/client` SDK:

 ```swift
 import Litebase

 let client = createClient(url: URL(string: "http://127.0.0.1:3000")!)
 let tokens = try await client.auth.signUp(email: "a@b.com", password: "secret12")
 let list = try await client.from("todos").select(
     where: ["completed": false],
     limit: 20
 )
 let unsub = try await client.from("todos").subscribeReady { event in
     print(event.op, event.data as Any)
 }
 unsub()
 client.closeRealtime()
 ```
 */

// Public types are defined across the module; this file is the package entry documentation.
